connectors/spark/src/test/java/io/unitycatalog/spark/BaseTableReadWriteTest.java

```diff
@@ -6,58 +6,206 @@
+      return this;
+    }
+
+    public SetupTableOptions setAsSelect(int i, String s) {
+      asSelect = Optional.of(Pair.of(i, s));
+      return this;
+    }
+
+    public SetupTableOptions setComment(String c) {
+      comment = Optional.of(c);
+      return this;
+    }
+
+    public String partitionClause() {
+      return partitionColumn.map(c -> String.format("PARTITIONED BY (%s)", c)).orElse("");
+    }
+
+    public String columnsClause() {
+      if (asSelect.isEmpty()) {
+        return "(i INT, s STRING)";
+      } else {
+        // CTAS can't specify columns
+        return "";
+      }
+    }
+
+    public String asSelectClause() {
+      return asSelect
+          .map(x -> String.format("AS SELECT %d AS i, '%s' AS s", x.getLeft(), x.getRight()))
+          .orElse("");
+    }
+
+    public String commentClause() {
+      return comment.map(c -> String.format("COMMENT '%s'", c)).orElse("");
+    }
+
+    public String createManagedTableSql() {
+      return String.format(
+          "CREATE TABLE %s.%s.%s %s USING %s %s %s %s %s",
+          catalogName,
+          schemaName,
+          tableName,
+          columnsClause(),
+          tableFormat(),
+          partitionClause(),
+          TBLPROPERTIES_CATALOG_OWNED_CLAUSE,
+          commentClause(),
+          asSelectClause());
+    }
+
+    public String createExternalTableSql(String location) {
+      return String.format(
+          "CREATE TABLE %s.%s.%s%s USING %s %s %s LOCATION '%s' %s",
+          catalogName,
+          schemaName,
+          tableName,
+          columnsClause(),
+          tableFormat(),
+          partitionClause(),
+          commentClause(),
+          location,
+          asSelectClause());
+    }
+
+    public String createDeltaPathTableSql(String location) {
+      assert testingDelta();
+      return String.format(
+          "CREATE TABLE %s.`%s`%s USING %s %s %s %s",
+          tableFormat(),
+          location,
+          columnsClause(),
+          tableFormat(),
+          partitionClause(),
+          commentClause(),
+          asSelectClause());
+    }
+
+    public String fullTableName() {
+      return String.join(".", catalogName, schemaName, tableName);
+    }
+  }
 
   /**
    * Set up a delta table and returns the table full name. This function is used by tests that
    * aren't cloud aware. It simply uses the same cloud as configured for managed storage.
    */
-  protected final String setupDeltaTable(
-      String catalogName, String tableName, List<String> partitionColumns)
-      throws IOException, ApiException {
-    return setupDeltaTable(managedStorageCloudScheme(), catalogName, tableName, partitionColumns);
+  protected final String setupTable(String catalogName, String tableName) {
+    return setupTable(new SetupTableOptions().setCatalogName(catalogName).setTableName(tableName));
+  }
+
+  /** Set up a delta table with storage on emulated cloud and returns the table full name. */
```

Comment:
is this still delta?
the method name does not clarify. if its to be used only for Delta then please keep it in the method name. if its for any format, how does this specify the format?